hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -401,6 +406,49 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		// This new condition is necessary for OCM personnel to report any cloud dangling objects to the user.
+		// The grace period is customizable using the flag --destroy-grace-period. It's a time.Duration field type.
+		// This condition only will be created when !SelfManage and CloudResourcesDestroyedCondition is False.
+		CloudResourcesDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+		if CloudResourcesDestroyedCondition.Status == "False" {
+			// ToDo (jparrill): when you deploy/destroy the cluster form CLI, you need to known the HCDestroyGracePeriod if not, on destroy
+			// the timeout will avoid to terminate the cluster properly. So we need to add an annotation with the HCDestroyGracePeriod for each
+			// HC deployed when tha value is > 0.
+			if r.HCDestroyGracePeriod > 0 {
+				reportingGracePeriod := &metav1.Condition{
+					Type:               string(hyperv1.HostedClusterDestroyed),
+					Status:             metav1.ConditionFalse,
+					Reason:             hyperv1.Done,
+					LastTransitionTime: metav1.Now(),
```

Comment:
you don't need to set `LastTransitionTime`, it is added automatically.

Also, it doesn't make sense for the reason to say `Done` when the status is `False`